eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/ImpTest.hs

```diff
@@ -152,7 +153,7 @@ addCollateralInput ::
   ImpTestM era (Tx era)
 addCollateralInput tx = impAnn "addCollateralInput" $ do
   ctx <- impGetPlutusContexts tx
```

Comment:
I've made this change, but there's still an `if null ctx` in the `otherwise` equation, to avoid adding collateral when there aren't any scripts. I assume this fits with what you had in mind.